### ISSUE TITLE
Усечение типов для выражения с Неопределено

### ISSUE THREAD

**Issue by RedMammoth** [1C-Company/v8-code-style#1243]
### Описание проблемы

Сейчас для того чтобы работал механизм усечения типов нужно в выражении условного оператора проверять тип, однако, как мне кажется, в случае с Неопределено чаще всего Тип опускают.

```bsl
// @strict-types

#Область ПрограммныйИнтерфейс

Процедура Тест() Экспорт
	
	СоставнойТип = Неопределено; // Неопределено, Число, Строка
	
	Если СоставнойТип <> Неопределено Тогда
		
		УсечениеНеРаботает = СоставнойТип;
		
	КонецЕсли;

	Если СоставнойТип = Неопределено Тогда
		
		УсечениеНеРаботает = СоставнойТип;
		
	КонецЕсли;
		
	Если ТипЗнч(СоставнойТип) <> Тип("Неопределено") Тогда
		
		УсечениеРаботает = СоставнойТип;
		
	КонецЕсли;
	
	Если ТипЗнч(СоставнойТип) = Тип("Неопределено") Тогда
		
		УсечениеРаботает = СоставнойТип;
		
	КонецЕсли;
	
КонецПроцедуры

#КонецОбласти
```

![image](https://user-images.githubusercontent.com/64589646/215186121-d72ae852-0a04-4e52-ad64-f81e47ff3d05.png)

![image](https://user-images.githubusercontent.com/64589646/215186195-c43c5e70-3edf-4585-bf68-239f96603b2b.png)

![image](https://user-images.githubusercontent.com/64589646/215186240-3c14d5f1-a968-4a58-befe-9371728433b3.png)

![image](https://user-images.githubusercontent.com/64589646/215186337-ebc35402-d1f1-4c45-bac5-af1cfb8d28ec.png)

### Описание решения проблемы

Предлагаю сделать усечение типов для варианта выражения с Неопределено также как и для Тип("Неопределено")

**Comment by artbear** [comment]
да, проверку на Неопределено однозначно нужно добавить для усечения типа

**Comment by RedMammoth** [comment]
Дополнение: было бы здорово, если бы усечение еще понимало выражение ЗначениеЗаполнено.
ЗначениеЗаполнено() всегда ложно для Неопределено и null и всегда истинно для непустых контейнеров:

![image](https://user-images.githubusercontent.com/64589646/219734496-e503413e-f2ce-4f2d-b1e5-5b2bbd7a27e7.png)
